_pytest/main.py

```diff
@@ -208,6 +210,24 @@ def pytest_ignore_collect(path, config):
     return False
 
 
+def pytest_collection_modifyitems(items, config):
+    deselect_prefixes = config.getoption("deselect")
+    if deselect_prefixes is None:
+        return
+
+    remaining = []
+    deselected = []
+    for colitem in items:
+        if colitem.nodeid.startswith(tuple(deselect_prefixes)):
```

Comment:
well done, please also move the call to `tuple` to the place where `deselect_prefixes` is declared